improver/utilities/field_texture.py

```diff
@@ -0,0 +1,208 @@
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Module containing plugin to calculate the texture in a field considering edge transitions."""
+
+import iris
+import numpy as np
+from iris.exceptions import CoordinateNotFoundError
+
+from improver import BasePlugin
+from improver.metadata.constants.attributes import MANDATORY_ATTRIBUTE_DEFAULTS
+from improver.metadata.utilities import create_new_diagnostic_cube
+from improver.nbhood.square_kernel import SquareNeighbourhood
+from improver.threshold import BasicThreshold
+from improver.utilities.cube_manipulation import collapsed
+
+
+class FieldTexture(BasePlugin):
+    """Plugin to calculate the texture in a binary field considering edge transitions
+
+    1) Takes a binary field that has been thresholded and looks for the transitions/edges
+       in the field that mark out a transition.
+    2) The transition calculation is then fed into the neighbourhooding code
+       (_calculate_ratio) to calculate a ratio for each cell.
+    3) The new cube of ratios is then thresholded and the realization coordinates
+       are collapsed to generate a mean of the thresholded ratios.
+    """
+
+    def __init__(self, nbhood_radius, ratio_threshold):
+        """
+
+        Args:
+            nbhood_radius (float):
+                The neighbourhood radius in metres within which the number of potential
+                transitions should be calculated. This forms the denominator in the
+                calculation of the ratio of actual to potential transitions that indicates a
+                field's texture. A larger radius should be used for diagnosing larger-scale
+                textural features.
+
+            ratio_threshold (float):
+                A unit-less threshold value that defines the ratio value above which
+                the field is considered clumpy and below which the field is considered
+                more contiguous.
```

Comment:
I think you could use clearer language here - "rough" or "smooth" texture.  But that's less important than keeping the terms consistent throughout the plugin.  So I'd pick some language for the top level docstring and then just check it's the same here.

Reply:
Chosen rough and smooth as suggested. This was most consistent throughout. Done.